### ISSUE TITLE
chain with string array causes "blue screen"

### ISSUE THREAD

**Issue by jjjordan** [robhagemans/pcbasic#74]
(Aside: This is awesome and I applaud your efforts)

Here's a short repro:

`b1.bas`
```vb
10 COMMON ARRAY$(), COUNT
15 DIM ARRAY$(10)
20 ARRAY$(1)="A"
25 ARRAY$(2)="B"
30 ARRAY$(3)="C"
35 COUNT=3
40 CHAIN "B2"
```

`b2.bas`
```vb
10 FOR I=1 TO COUNT
15  PRINT ARRAY$(I)
20 NEXT
```

A quick glance at the log seems to indicate that it's not copying over the last element of the array in the CHAIN.  If I change `COUNT` to 2 then this works.  But if I also remove line 30 then it goes back to crashing.  The log is missing the last element's string in both cases.

```
==== Scalars =======================================================================================
COUNT!: ![00004082 3.0]
I!: ![00004082 3.0]
==== Arrays ========================================================================================
ARRAY$[10]: 00fdfd01fcfd01fbfd01fafd00fafd00fafd00fafd00fafd00fafd00fafd00fafd
==== Strings =======================================================================================
fdfb: bytearray(b'B')
fdfc: bytearray(b'A')
==== Program Buffer ================================================================================
00 8012 (+018) 0a00 [00010] 822049e71220cc20434f554e54
00 9112 (+017) 0f00 [00015] 209120415252415924284929
00 9712 (+006) 1400 [00020] 83
00 0000 (ENDS)
==== Program =======================================================================================
10 FOR I=1 TO COUNT
15  PRINT ARRAY$(I)
20 NEXT
==== Options =======================================================================================
[]
```

Finally, here's the traceback
```
Traceback (most recent call last):
  File "/home/jj/src/jbc/venv/local/lib/python2.7/site-packages/pcbasic/guard.py", line 48, in protect
    yield
  File "/home/jj/src/jbc/venv/local/lib/python2.7/site-packages/pcbasic/main.py", line 110, in run_session
    session.interact()
  File "/home/jj/src/jbc/venv/local/lib/python2.7/site-packages/pcbasic/basic/api.py", line 115, in interact
    self._impl.interact()
  File "/home/jj/src/jbc/venv/local/lib/python2.7/site-packages/pcbasic/basic/implementation.py", line 282, in interact
    self.interpreter.loop()
  File "/home/jj/src/jbc/venv/local/lib/python2.7/site-packages/pcbasic/basic/interpreter.py", line 120, in loop
    self.parse()
  File "/home/jj/src/jbc/venv/local/lib/python2.7/site-packages/pcbasic/basic/interpreter.py", line 109, in parse
    self.parser.parse_statement(ins)
  File "/home/jj/src/jbc/venv/local/lib/python2.7/site-packages/pcbasic/basic/parser/statements.py", line 81, in parse_statement
    self._callbacks[c](parse_args(ins))
  File "/home/jj/src/jbc/venv/local/lib/python2.7/site-packages/pcbasic/basic/devices/files.py", line 424, in print_
    formatter.Formatter(output, screen).format(args)
  File "/home/jj/src/jbc/venv/local/lib/python2.7/site-packages/pcbasic/basic/devices/formatter.py", line 39, in format
    self._print_value(next(args))
  File "/home/jj/src/jbc/venv/local/lib/python2.7/site-packages/pcbasic/basic/devices/formatter.py", line 52, in _print_value
    word = expr.to_str()
  File "/home/jj/src/jbc/venv/local/lib/python2.7/site-packages/pcbasic/basic/values/strings.py", line 39, in dereference
    return self._stringspace.view(length, address).tobytes()
  File "/home/jj/src/jbc/venv/local/lib/python2.7/site-packages/pcbasic/basic/values/strings.py", line 190, in view
    return memoryview(self._retrieve(length, address))
  File "/home/jj/src/jbc/venv/local/lib/python2.7/site-packages/pcbasic/basic/values/strings.py", line 181, in _retrieve
    return bytearray() if length == 0 else self._strings[address]
KeyError: 65018
```

**Comment by robhagemans** [comment]
Thanks for the report and the example code - I can reproduce this and will investigate.

**Comment by robhagemans** [comment]
Fixed by commit b9e1118, will be in next release.
